File: hexchat/highlight.pl
use strict;

Xchat::register("Highlight Logger", "1.0");
Xchat::print("Highlight Logger loaded\n");

my $tabName = '(Highlights)';

sub highlightHandler {
	my $channel = Xchat::get_info('channel');
	if ($_[1] eq 'Channel Msg Hilight') {
		Xchat::print("$channel\t\00320<$_[0][3]$_[0][2]$_[0][0]> $_[0][1]", $tabName);
	}
	elsif ($_[1] eq 'Channel Action Hilight') {
		Xchat::print("$channel\t\00320$_[0][3]$_[0][2]$_[0][0] $_[0][1]", $tabName);
	}
	return Xchat::EAT_NONE;
}

Xchat::hook_print('Channel Msg Hilight', \&highlightHandler, { data => 'Channel Msg Hilight' });
Xchat::hook_print('Channel Action Hilight', \&highlightHandler, { data => 'Channel Action Hilight' });

Xchat::command("query $tabName");

File: hexchat/highlight.py
import hexchat

__module_name__ = 'Highlight Logger'
__module_version__ = '1.0'
__module_description__ = ''

hexchat.prnt('Highlight Logger loaded')

tab_name = '(Highlights)'

def highlight_callback(word, word_eol, user_data):
	word = [(word[i] if len(word) > i else '') for i in range(4)]
	
	channel = hexchat.get_info('channel')
	highlight_context = hexchat.find_context(channel=tab_name)
	
	if user_data == 'Channel Msg Hilight':
		highlight_context.prnt('{0}\t\00320<{4}{3}{1}> {2}'.format(channel, word[0], word[1], word[2], word[3]))
	elif user_data == 'Channel Action Hilight':
		highlight_context.prnt('{0}\t\00320{4}{3}{1} {2}'.format(channel, word[0], word[1], word[2], word[3]))
	
	return hexchat.EAT_NONE

hexchat.hook_print('Channel Msg Hilight', highlight_callback, 'Channel Msg Hilight')
hexchat.hook_print('Channel Action Hilight', highlight_callback, 'Channel Action Hilight')

hexchat.command('query {0}'.format(tab_name))
